vignettes/datatable-joins.Rmd

```diff
@@ -700,25 +700,77 @@ Products[!"popcorn",
 
 ### 7.2. Updating by reference
 
-The `:=` operator in `data.table` is used for updating or adding columns by reference. This means it modifies the original `data.table` without creating a copy, which is very memory-efficient, especially for large datasets. When used inside a `data.table`, `:=` allows you to **add new columns** or **modify existing ones** as part of your query.
+Use `:=` to modify columns **by reference** (no copy) during joins. General syntax: `x[i, on=, (cols) := val]`.
 
-Let's update our `Products` table with the latest price from `ProductPriceHistory`:
+**Simple One-to-One Update**  
+Update `Products` with prices from `ProductPriceHistory`:
 
 ```{r}
-copy(Products)[ProductPriceHistory,
-               on = .(id = product_id),
-               j = `:=`(price = tail(i.price, 1),
-                        last_updated = tail(i.date, 1)),
-               by = .EACHI][]
+Products[ProductPriceHistory, 
+         on = .(id = product_id), 
+         price := i.price]
 ```
+- `i.price` refers to price from `i` `(ProductPriceHistory)`.
+- Modifies Products in-place.
 
-In this operation:
+**Grouped Updates with `.EACHI`**
+Get last price/date for each product:
+```{r Updating_with_the_Latest_Record}
+Products[ProductPriceHistory,
+         on = .(id = product_id),
+         `:=`(price = last(i.price), last_updated = last(i.date)),
+         by = .EACHI]
+```
+- `by = .EACHI` groups by i's rows (1 group per Products row).
+- `last()` returns last value including `NA`:
+
+```{r}
+data.table::last(c(1, NA))  # NA
+```
+**Efficient Right Join Update**
+Add product details to `ProductPriceHistory` without copying:
+
+```{r}
+cols <- setdiff(names(Products), "id")
+ProductPriceHistory[, (cols) := 
+  Products[.SD, on = .(id = product_id), .SD, .SDcols = cols]]
+```
+- `.SD` refers to `ProductPriceHistory` during the join.
+- Updates `ProductPriceHistory` by reference.
+
+**Handling Edge Cases and Dynamic Column Updates**
+To dynamically update columns and handle missing values:
+```{r}
+cols <- setdiff(names(Products), "id")
+ProductPriceHistory[, (cols) := 
+  Products[.SD, on = .(id = product_id), .SD, .SDcols = cols]]
+ProductPriceHistory[is.na(price), price := 0]  # Handle missing values
+```
+- Ensures unmatched values do not propagate `NA` unintentionally.
+
+**Dynamic Column Selection and Updates**
+Columns can be dynamically updated based on variable names:
+```{r}
+my_var_name <- "price"
+Products[ProductPriceHistory, on = .(id = product_id), 
+         (my_var_name) := i.price]
+```
+- This approach allows flexibility in specifying columns programmatically.
```

Comment:
The examples in https://stackoverflow.com/a/44592473 include what I believe is another interesting point, that the `on =` option may also depend on the `my_var_name` (`on = paste0(my_var_name, "==from")`)